corehq/apps/userreports/indicators/factory.py

```diff
@@ -85,6 +87,42 @@ def _construct_column(choice):
     return CompoundIndicator(base_display_name, choice_indicators)
 
 
+def _build_ledger_balances_indicator(spec, context):
+    wrapped_spec = LedgerBalancesIndicatorSpec.wrap(spec)
+    base_display_name = wrapped_spec.display_name
+
+    def _construct_display(product_code):
+        return '{base} ({product_code})'.format(base=base_display_name,
+                                                product_code=product_code)
+
+    def _construct_column(product):
+        return '{col}_{product}'.format(col=spec['column_id'], product=product)
+
+    def _make_get_stock_state(product_code):
+        def get_stock_state(doc, context):
+            try:
+                return StockState.objects.get(
+                    section_id=wrapped_spec.ledger_section,
+                    case_id=doc['_id'],
+                    sql_product__code=product_code,
+                ).stock_on_hand
+            except StockState.DoesNotExist:
+                return 0
+        return get_stock_state
+
+    product_indicators = [
+        RawIndicator(
+            display_name=_construct_display(product_code),
+            column=Column(
+                id=_construct_column(product_code),
+                datatype="decimal",
+            ),
+            getter=_make_get_stock_state(product_code),
```

Comment:
I saw some other indicator types used a`getter` property on the `IndicatorSpec`.  The problem is this needs to know the product_code, and I'm not sure how else to pass it through.  The other thing I'm thinking of (but haven't yet implemented) is reducing the db hits here - a simple optimization is:
```python
# switch from
StockState.objects.get(
    section_id=wrapped_spec.ledger_section,
    case_id=doc['_id'],
    sql_product__code=product_code,
)
# to
StockState.objects.filter(
    section_id=wrapped_spec.ledger_section,
    case_id=doc['_id'],
    sql_product__code__in=product_codes,
)
```
Such an optimization would require passing in shared state to the getter like I've done here with `product_code`.